src/smtp-lib/socket_listener.cpp

```diff
@@ -0,0 +1,12 @@
+#include "socket_listener.hpp"
+
+#include "connection.hpp"
+
+SocketListener::SocketListener(const uint16_t port) : m_port(port), m_raw_socket(RawSocket::new_socket()) {
+    m_raw_socket.bind(m_port);
+    m_raw_socket.listen(5);
```

Comment:
doesn't this mean we can only have 5 connections? maybe we should change this

Reply:
Yes, not part of this merge request.